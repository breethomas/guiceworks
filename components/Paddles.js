// @flow
import React from 'react'
import css, { media2 } from '../styles/css'
import Header from './Header'
import Link from './Link'
import { ArrowIcon } from './Icons'

type Props = {
  dir: string,
  nextLink: string,
  prevLink: string,
}

const style = css(
  {
    color: '#010101',
    display: 'flex',
    flexFlow: 'row wrap',
    height: 32,
    marginBottom: 20,
  },
  media2({ marginBottom: 80 }),
)

const nextLinkStyle = css({
  marginLeft: 'auto',
})

export default ({ dir, nextLink, prevLink }: Props) => (
  <Header className={style} >
    <Link
      as={`/${dir}/${prevLink}`}
      href={`/${dir}?id=${prevLink}`}
      title={prevLink}
    >
      <ArrowIcon style={{ transform: 'rotate(180deg)' }} />
    </Link>
    <Link
      as={`/${dir}/${nextLink}`}
      href={`/${dir}?id=${nextLink}`}
      className={nextLinkStyle}
      title={nextLink}
    >
      <ArrowIcon />
    </Link>
  </Header>
)

