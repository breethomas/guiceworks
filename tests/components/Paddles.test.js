import React from 'react'
import renderer from 'react-test-renderer'
import Paddles from '../../components/Paddles'

const noop = () => {}

describe('Paddles', () => {
  let component

  beforeEach(() => {
    component = (
      <Paddles
        dir="projects"
        nextLink={noop}
        prevLink={noop}
      />
    )
  })

  it('shows the correct snapshot for the Paddles component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
