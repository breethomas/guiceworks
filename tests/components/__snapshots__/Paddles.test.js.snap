// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paddles shows the correct snapshot for the Paddles component tree 1`] = `
<header
  className={
    Object {
      "data-css-1f6b3ur": "",
    }
  }
>
  <a
    className={
      Object {
        "data-css-33otqm": "",
      }
    }
    href="/projects/function noop() {}"
    onClick={[Function]}
    title={[Function]}
  >
    <svg
      aria-hidden={true}
      height={32}
      style={
        Object {
          "transform": "rotate(180deg)",
        }
      }
      viewBox="0 0 32 32"
      width={32}
    >
      <g
        fill="currentColor"
      >
        <polygon
          points="15.9 0 13.725 2.163 26.11 14.472 0 14.472 0 17.529 26.11 17.529 13.725 29.839 15.9 32 32 16"
        />
      </g>
    </svg>
  </a>
  <a
    className="css-33otqm css-omocl"
    href="/projects/function noop() {}"
    onClick={[Function]}
    title={[Function]}
  >
    <svg
      aria-hidden={true}
      height={32}
      viewBox="0 0 32 32"
      width={32}
    >
      <g
        fill="currentColor"
      >
        <polygon
          points="15.9 0 13.725 2.163 26.11 14.472 0 14.472 0 17.529 26.11 17.529 13.725 29.839 15.9 32 32 16"
        />
      </g>
    </svg>
  </a>
</header>
`;
